src/BenchmarkDotNet.Core/Columns/SizeUnit.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BenchmarkDotNet.Columns
+{
+    public class SizeUnit
+    {
+        public string Name { get; }
+        public string Description { get; }
+        public long ByteAmount { get; }
+
+        public SizeUnit(string name, string description, long byteAmount)
+        {
+            Name = name;
+            Description = description;
+            ByteAmount = byteAmount;
+        }
+
+        public const long BytesInKiloByte = 1000L;
+        public static readonly SizeUnit B = new SizeUnit("B", "Byte", 1L);
+        public static readonly SizeUnit kB = new SizeUnit("kB", "Kilobyte", BytesInKiloByte);
+        public static readonly SizeUnit MB = new SizeUnit("MB", "Megabyte", BytesInKiloByte * BytesInKiloByte);
+        public static readonly SizeUnit GB = new SizeUnit("GB", "Gigabyte", BytesInKiloByte * BytesInKiloByte * BytesInKiloByte);
+        public static readonly SizeUnit TB = new SizeUnit("TB", "Terabyte", BytesInKiloByte * BytesInKiloByte * BytesInKiloByte * BytesInKiloByte);
+        public static readonly SizeUnit[] All = { B, kB, MB, GB, TB };
+
+        public static SizeUnit GetBestSizeUnit(params long[] values)
```

Comment:
By the way, I have a similar feature request: https://github.com/dotnet/BenchmarkDotNet/issues/404 I think it makes sense to add this feature right now because it's also a common logic for `IMeasurementUnit`. Could you implement it?